dashboard-frontend/src/server.ts

```diff
@@ -80,6 +148,101 @@ export const makeServer = ({ environment = 'development' } = {}) => {
           },
         };
       });
+
+      this.get('/studies/:slug', ({ db }, request) => {
+        const study = (db.studies as any).findBy({ slug: request.params.slug });
+
+        return {
+          data: study,
+        };
+      });
+
+      /**
+       * Things to consider when implementing the real endpoint:
+       *  - When a Study is initially created, an associated StudyProgress must be created with default values.
```

Comment:
I don't like this as a requirement. A study that has been created but doesn't yet have any progress seems like a valid database state and I think enforcing that to be invalid is brittle. It might be useful to create an initial "studyprogress" with the progress at time 0 when a study has been created, but that shouldn't be a _requirement_ for the frontend. 

The frontend should therefore be able to handle the state where there is no studyprogress. That could mean that the whole page is replaced by text saying "Nothing to report yet! Check back soon", which would be fine.

Reply:
I agree, good point 👌

I created an issue to handle this: https://github.com/vlab-research/vlab/issues/31

Reply:
Cool, thanks!